### PR TITLE
impl new version user & tag search

### PR DESCRIPTION
**Pull request by gary02**
- fix db setup script for tests with native database 
- preprocess searchV1 input (t2s, lowercase, strip spaces) 
- impl `UserService.searchV1`
- impl `TagService.SearchV1`

resolve https://github.com/thematters/matters-server/issues/2971

### CODE REVIEW

File: src/connectors/tagService.ts [hunk 1]
```diff
@@ -685,7 +684,48 @@ export class TagService extends BaseService {
...
+    if (!_key) {
+      return { nodes: [], totalCount: 0 }
+    }
+
+    const mattyChoiceTagIds = environment.mattyChoiceTagId
+      ? [environment.mattyChoiceTagId]
+      : []
+
+    // search from original tags but return duplicate free tags
+    const queryIds = this.knex('search_index.tag')
```
Comment (robertu7) [inline src/connectors/tagService.ts:699]: For performance reason, shall we `.limit` the query results?

File: src/connectors/userService.ts [hunk 2]
```diff
@@ -600,7 +600,49 @@ export class UserService extends BaseService {
...
+    }
+
+    const baseQuery = this.knex('search_index.user')
+      .select('id')
+      .whereLike('display_name', `%${displayName}%`)
+      .orWhereLike('user_name', `%${userName}%`)
+      .orderByRaw('display_name = ? DESC', [displayName])
+      .orderByRaw('user_name = ? DESC', [userName])
+      .orderByRaw('display_name ~ ? DESC', [displayName])
+      .orderByRaw('num_followers DESC NULLS LAST')
```
Comment (robertu7) [inline src/connectors/userService.ts:618]: Same question as above, shall we `.limit` the query results?

**Review by robertu7 — APPROVED** [review]

File: src/connectors/tagService.ts (hunk 1 above)
Comment (gary02) [reply src/connectors/tagService.ts:699]: limit do be set in L723？

File: src/connectors/userService.ts (hunk 2 above)
Comment (gary02) [reply src/connectors/userService.ts:618]: limit is set in L639

File: src/connectors/tagService.ts (hunk 1 above)
Comment (robertu7) [reply src/connectors/tagService.ts:699]: But the query performance of L702 impacted by `queryIds` right?

File: src/connectors/userService.ts [hunk 3]
```diff
@@ -600,7 +600,49 @@ export class UserService extends BaseService {
...
+
+    if (!userName) {
+      return { nodes: [], totalCount: 0 }
+    }
+
+    const baseQuery = this.knex('search_index.user')
+      .select('id')
+      .whereLike('display_name', `%${displayName}%`)
+      .orWhereLike('user_name', `%${userName}%`)
+      .orderByRaw('display_name = ? DESC', [displayName])
```
Comment (tx0c) [inline src/connectors/userService.ts:615]: here need to check if search key startsWith('@'), then search on user_name only, no matching on `display_name`

File: src/connectors/userService.ts [hunk 4]
```diff
@@ -600,7 +600,49 @@ export class UserService extends BaseService {
...
+      .orderByRaw('user_name = ? DESC', [userName])
+      .orderByRaw('display_name ~ ? DESC', [displayName])
```
Comment (tx0c) [inline src/connectors/userService.ts:617]: and here, if startsWith('@') then no check display_name

File: src/connectors/tagService.ts (hunk 1 above)
Comment (gary02) [reply src/connectors/tagService.ts:699]: Yes， but if `queryIds` is limited,  the final search result will lack some hit items?

**Pull request merged**